varats-core/varats/mapping/author_map.py

```diff
@@ -0,0 +1,165 @@
+
+from varats.utils.git_util import __get_git_path_arg
+
+LOG = logging.getLogger(__name__)
+
+
+class AmbiguousAuthor(Exception):
+    """Raised if an ambiguous author is encountered."""
+
+
+NAME_REGEX = re.compile("\s*\d+\t(.*) <(.*)>")
+
+
+class Author():
+    """Representation of one author."""
+
+    def __init__(self, id: int, name: str, email: str) -> None:
+        self.id = id
+        self.name = name
+        self.__names = {name}
+        self.mail = email
+        self.__mail_addresses = {email}
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, Author):
+            return self.id == other.id
+        return False
+
+    def __str__(self) -> str:
+        return f"{self.id} {self.name} <{self.mail}>"
+
+    def __repr__(self) -> str:
+        return f"{self.id} {self.name} <{self.mail}>; {self.names},{self.mail_addresses}"
+
+    @property
+    def names(self) -> tp.Set[str]:
+        return self.__names
+
+    @property
+    def mail_addresses(self) -> tp.Set[str]:
+        return self.__mail_addresses
+
+    def add_data(self, name: str, mail: str) -> None:
+        if not name in self.names:
+            self.names.add(name)
+        if not mail in self.mail:
+            self.mail_addresses.add(mail)
+
+    def merge(self, other: 'Author') -> 'Author':
+        if other.id < self.id:
+            other.names.union(self.names)
+            other.mail_addresses.union(self.mail_addresses)
+            return other
+
+        self.names.union(other.names)
+        self.mail_addresses.union(other.mail_addresses)
+        return self
+
+
+class AuthorMap():
+    """Provides a mapping of an author to all combinations of author name to
+    mail address."""
+
+    def __init__(self) -> None:
+        self._look_up_invalid = True
+        self.current_id = 0
+        self.mail_dict: tp.Dict[str, Author] = {}
+        self.name_dict: tp.Dict[str, Author] = {}
+        self.__authors: tp.List[Author] = []
+
+    def get_author_by_name(self, name: str) -> tp.Optional[Author]:
+        if self._look_up_invalid:
+            self._gen_lookup_dicts()
+        return self.name_dict.get(name, None)
+
+    def get_author_by_email(self, email: str) -> tp.Optional[Author]:
+        if self._look_up_invalid:
+            self._gen_lookup_dicts()
+        return self.mail_dict.get(email, None)
+
+    @property
+    def authors(self) -> tp.List[Author]:
+        return self.__authors
+
+    def get_author(self, name: str, mail: str) -> tp.Optional[Author]:
+        """
+        Get an author by name and mail.
+
+        Returns None if no author or multiple authors were found matching the
+        combination of name and mail-address.
+        """
+        if self._look_up_invalid:
+            self._gen_lookup_dicts()
+        mail_author = self.mail_dict.get(mail, None)
+        if mail_author == (name_author := self.name_dict.get(name, None)):
+            return name_author
+        else:
+            return None
```

Comment:
```suggestion

        return None
```
@Sinerum You don't need an else if you return within the then clause